controllers/autohealconfig_controller.go

```diff
@@ -47,16 +66,415 @@ type AutohealConfigReconciler struct {
-	_ = log.FromContext(ctx)
+	autoheal := &v1alpha1.AutohealConfig{}
+	err := r.Client.Get(ctx, req.NamespacedName, autoheal)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Infof("Autoheal CR not found%s", err)
+			return ctrl.Result{}, nil
+		}
+		log.Errorf("Error getting Autoheal CR %s", err)
+		return ctrl.Result{}, err
+	}
+
+	if autoheal.DeletionTimestamp == nil && autoheal.Finalizers == nil {
+		autoheal.Finalizers = append(autoheal.Finalizers, AutohealConfigfinalizer)
+		err = r.Update(ctx, autoheal)
+		if err != nil {
+			log.Errorf("Unable to add finalizer %s", err)
+			return ctrl.Result{}, err
+		}
+	}
+
+	if autoheal.DeletionTimestamp == nil {
+		if autoheal.Status.LastSyncPeriod != 0 {
+			lastSync := time.Unix(autoheal.Status.LastSyncPeriod, 0)
+			period, err := time.ParseDuration(autoheal.Spec.SyncPeriod)
+			if autoheal.Spec.SyncPeriod == "" {
+				log.Info("Please specify a sync period")
+			}
+			if err != nil {
+				log.Errorf("Error parsing operator resync period %s", err)
+				return ctrl.Result{}, err
+
+			}
+			nextSync := lastSync.Add(period)
+			if time.Now().Before(nextSync) {
+				return ctrl.Result{
+					Requeue:      true,
+					RequeueAfter: RequeueDelaySuccess,
+				}, nil
+
+			}
+		}
+		configRepo, err := r.GetConfigRepoSecret(ctx, autoheal, &autoheal.Status)
+		if err != nil {
+			log.Errorf("Unable to reconcile getting config repo secret %s", err)
+		}
+		files, err := ReadFiles(configRepo)
+		if err != nil {
+			log.Errorf("Unable to get file endpoints from config repo %s", err)
+		}
+		scenarios, err := r.getScenarios(files, configRepo, autoheal, ctx)
+		if err != nil {
+			log.Errorf("Unable to get scenerios from config repo %s", err)
+
+		}
+		err = r.createConfigMaps(autoheal, scenarios, ctx)
+		if err != nil {
+			log.Errorf("Unable to create config maps %s", err)
+		}
+		err = r.removeUnwantedCM(ctx, &scenarios)
+		if err != nil {
+			log.Error("Unable to delete unwanted configs")
+		}
+	} else {
+		err = r.deleteConfigmap(autoheal, ctx)
+		if err != nil {
+			log.Errorf("Unable to reconcile deleting config maps %s", err)
+		}
 
-	// TODO(user): your logic here
+		err = r.deleteConfigSecret(autoheal, ctx)
+		if err != nil {
+			log.Errorf("Unable to reconcile deleting config secret %s", err)
+		}
+		if autoheal.Finalizers != nil {
+			autoheal.Finalizers = []string{}
+			err = r.Update(ctx, autoheal)
+			if err != nil {
+				log.Errorf("Unable to update finalizer to remove it %s", err)
+				return ctrl.Result{}, err
+			}
+		}
+	}
 
-	return ctrl.Result{}, nil
+	return r.updateStatus(autoheal)
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *AutohealConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&autohealv1alpha1.AutohealConfig{}).
+		For(&v1alpha1.AutohealConfig{}).
 		Complete(r)
 }
+
+// Gets the secret needed to provide credentials to sycn with github repo
```

Comment:
```suggestion
// GetConfigRepoSecret gets the secret needed to provide credentials to sync with the GitHub repo
```
This is something my IDE flagged again: comments describing a function should begin with the function name (I think this is best practice).

Reply:
ok cool let me update those